src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -115,33 +115,108 @@ public static void Analyze(CSharpCompilation compilation, Symbol member, BoundNo
 
         protected override void Normalize(ref LocalState state)
         {
+            int oldNext = state.Capacity;
             state.EnsureCapacity(nextVariableSlot);
+            for (int i = oldNext; i < nextVariableSlot; i++)
+            {
+                var id = variableBySlot[i];
+                // Initialize non-struct members with the declared nullability.
+                // (Struct members are handled by InheritNullableStateOfTrackableStruct.)
+                bool? isNotNull = null;
+                int slot = id.ContainingSlot;
+                if (slot > 0 && variableBySlot[slot].Symbol.GetTypeOrReturnType().TypeKind != TypeKind.Struct)
```

Comment:
>if (slot > 0 && variableBySlot[slot].Symbol.GetTypeOrReturnType().TypeKind != TypeKind.Struct) [](start = 16, length = 94)

I am not sure if looking just one level up is enough. What if it is in a struct accessed through a class?